### ISSUE TITLE
TS type inference is broken for pipes of exactly two operators

### ISSUE THREAD

**Issue by riimeik** [ReactiveX/IxJS#269]
**IxJS version:** 2.5.2

**Typescript version:** 3.3.3

**Expected behavior:**

Types should be properly inferred from the type of the iterable and the return types of pipe operators.

**Actual behavior:**

When the pipe is given exactly two arguments, type inference breaks down:

```
import { AsyncIterableX } from 'ix/asynciterable';
import * as IxOp from 'ix/asynciterable/pipe/';

AsyncIterableX.from([1, 2]).pipe(
    IxOp.map(num => num + 1),
    IxOp.delay(2)
);
```
This is the error that is reported:
![error](https://user-images.githubusercontent.com/13738028/54822584-fe3d6380-4cad-11e9-94ac-fc10a0b7b9d5.png).

Curiously, the issue is fixed when changing the sequence of overloads for the **pipe** method:
![fix](https://user-images.githubusercontent.com/13738028/54822585-fe3d6380-4cad-11e9-88f9-8469e0ea1501.png).

**Comment by trxcllnt** [comment]
@riimeik could you post those screenshots inline with the github issue? I'm getting `403 Forbidden` errors when I try to view them. I know we had to fix some things here in 2.5.2, but maybe I broke it in other ways.

**Comment by riimeik** [comment]
@trxcllnt Updated the issue.

**Comment by trxcllnt** [comment]
@riimeik this has been published in the 2.5.3 release
